_posts/en/newsletters/2025-04-11-newsletter.md

```diff
@@ -0,0 +1,212 @@
+  implements its virtual methods `TransactionAddedToMempool`,
+  `TransactionRemovedFromMempool`, and
+  `MempoolTransactionsRemovedForBlock`. This gives
+  `CBlockPolicyEstimator` all the mempool information it needs without
+  being unnecessarily tightly coupled to the mempool via a reference."
+  a1link="https://bitcoincore.reviews/31664#l-26"
+
+  q2="What are the trade-offs between the new architecture and a direct
+  modification of CBlockPolicyEstimator?"
+  a2="The new architecture with a `FeeRateForecasterManager` class to
+  which multiple `Forecaster`s can be registered is a more modular
+  approach which allows for better testing, and enforces a better
+  separation of concerns. It allows to easily plug in new forecasting
+  strategies later on. This comes at the cost of a bit more code to
+  maintain, and potentially confusing users about which estimation
+  method to use."
+  a2link="https://bitcoincore.reviews/31664#l-43"
+%}
+
+## Releases and release candidates
+
+_New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates._
+
+- [Bitcoin Core 29.0rc3][] is a release candidate for the next major
+  version of the network's predominate full node.  Please see the
+  [version 29 testing guide][bcc29 testing guide].
+
+- [LND 0.19.0-beta.rc2][] is a release candidate for this popular LN
+  node.  One of the major improvements that could probably use testing
+  is the new RBF-based fee bumping for cooperative closes.
+
+<!-- FIXME:harding to update Thursday -->
+
+## Notable code and documentation changes
+
+_Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
+[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
+repo], and [BINANAs][binana repo]._
+
+- [LDK #2256][] and [LDK #3709][] implements attributable failures (see
+  Newsletter [#224][news224 failures]) as specified in [BOLTs #1044][] by adding
+  an optional `attribution_data` field to the `UpdateFailHTLC` struct and
+  introducing the `AttributionData` struct. In this protocol, each forwarding
+  node appends to the failure message a `hop_payload` flag, a duration field
+  that records how long the node held the HTLC, and HMACs corresponding to
+  different assumed positions in the route. If a node corrupts the failure
+  message, the mismatch in the HMAC chain helps identify the pair of nodes
+  between which this happened. "In addition, tracking HTLC hold times could
+  allow a node to penalize a routing node for excessive delay.
```

Comment:
I couldn't find a source for this quote in a quick search of the four sources we linked (two LDK PRs, news224, BOLTs 1044) or a web search, so I'm going to remove it.  Sorry if the source is just staring me in the face.